lstchain/calib/camera/r0.py

```diff
@@ -0,0 +1,288 @@
+        """
+        if self.pedestal_path:
+            with fits.open(self.pedestal_path) as f:
+                n_modules = f[1].header['NAXIS4']
+                self.pedestal_value_array = np.zeros((n_modules, 2, 7, 4136), dtype=np.int16)
+                pedestal_data = np.int16(f[1].data)
+                self.pedestal_value_array[:, :, :, :self.size4drs] = pedestal_data - self.offset
+                self.pedestal_value_array[:, :, :, self.size4drs:self.size4drs + 40] \
+                    = pedestal_data[:, :, :, 0:40] - self.offset
+
+    def _get_first_capacitor(self, event, nr_module):
+        """
+        Get first capacitor values from event for nr module.
+        Parameters
+        ----------
+        event : `ctapipe` event-container
+        nr_module : number of module
+        """
+        fc = np.zeros((2, 7))
+        first_cap = event.lst.tel[0].evt.first_capacitor_id[nr_module * 8:
+                                                            (nr_module + 1) * 8]
+        for i, j in zip([0, 1, 2, 3, 4, 5, 6], [0, 0, 1, 1, 2, 2, 3]):
+            fc[self.high_gain, i] = first_cap[j]
+        for i, j in zip([0, 1, 2, 3, 4, 5, 6], [4, 4, 5, 5, 6, 6, 7]):
+            fc[self.low_gain, i] = first_cap[j]
+        return fc
+
+@jit(parallel=True)
+def subtract_pedestal_jit(event_waveform, expected_pixel_id, fc_cap, pedestal_value_array, n_modules):
+    waveform = np.zeros(event_waveform.shape)
+    size4drs = 4096
+    n_gain = 2
+    n_pix = 7
+    for nr in prange(0, n_modules):
+        for gain in prange(0, n_gain):
+            for pix in prange(0, n_pix):
+                pixel = expected_pixel_id[nr*7 + pix]
+                position = int((fc_cap[nr, gain, pix]) % size4drs)
+                waveform[gain, pixel, :] = \
+                    (event_waveform[gain, pixel, :] -
+                    pedestal_value_array[nr, gain, pix, position:position + 40])
+    return waveform
+
+@jit(parallel=True)
+def do_time_lapse_corr(waveform, expected_pixel_id, local_clock_list, fc, last_time_array, number_of_modules):
+    size4drs = 4096
+    for nr_clus in prange(0, number_of_modules):
+        time_now = local_clock_list[nr_clus]
+        for gain in prange(0, 2):
+            for pix in prange(0, 7):
+                pixel = expected_pixel_id[nr_clus*7 + pix]
+                for k in prange(0, 40):
+                    posads = int((k + fc[nr_clus, gain, pix]) % size4drs)
+                    if last_time_array[nr_clus, gain, pix, posads] > 0:
+                        time_diff = time_now - last_time_array[nr_clus, gain, pix, posads]
+                        val = waveform[gain, pixel, k] - ped_time(time_diff / (133.e3))
+                        waveform[gain, pixel, k] = val
+                    if (k < 39):
+                        last_time_array[nr_clus, gain, pix, posads] = time_now
+
+@jit
+def ped_time(timediff):
+    return 29.3 * np.power(timediff, -0.2262) - 12.4
+
```

Comment:
where are these numbers (29.3, 0.2262, 12.4) coming from? I would add a comment about those

Reply:
Numbers come from curve fitting. I added comment. Now, this is fixed numbers for all pixels, but I will investigate how those coefficients depends of pixel.